### PR TITLE
add openssl 3.0.5

### PR DESCRIPTION
**Pull request by SkyTeamWater**
add openssl 3.0.5

### CODE REVIEW

File: packages/o/openssl/xmake.lua [hunk 1]
```diff
@@ -4,21 +4,24 @@ package("openssl")
     set_description("A robust, commercial-grade, and full-featured toolkit for TLS and SSL.")
 
     add_urls("https://github.com/openssl/openssl/archive/refs/tags/OpenSSL_$(version).zip", {version = function (version)
-        return version:gsub("%.", "_")
-    end, excludes = "*/fuzz/*"})
-    add_versions("1.1.1q", "df86e6adcff1c91a85cef139dd061ea40b7e49005e8be16522cf4864bfcf5eb8")
+	return version:gsub("%.", "_")
+    end, excludes = "*/fuzz/*"}, {alias = "1"})
+    add_urls("https://github.com/openssl/openssl/archive/refs/tags/openssl-$(version).zip", version, {alias = "3"})
```
Comment (waruqi) [inline packages/o/openssl/xmake.lua:9]: 没必要用 alias 。。把 `OpenSSL_$(version).zip` 也整个放进 `$(version).zip` 然后在 `function (version) end` 里面拼下就行了。。

另外 add_patches 跟 add_urls  混一起有点乱，add_patches 统一放 add_urls 下面

**Comment by waruqi** [comment]
看了下，3 的接口都不一定兼容了，还是拆成独立包好了。。https://github.com/xmake-io/xmake-repo/pull/1375#issuecomment-1200590810

**Comment by AsakuraMizu** [comment]
so where is openssl3

**Comment by bradfordzhang** [comment]
It seems not yet, you can add it :)

**Comment by bradfordzhang** [comment]
> so where is openssl3

It seems not yet, you can add it :)

**Comment by waruqi** [comment]
you can add a new openssl3 package

**Pull request closed without merging**